helpers/helper.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+import grequests
+from jinja2 import Environment, FileSystemLoader
+import os
+from .config import (
+    LMS_URL,
+    STUDIO_URL,
+    INSTANCE_NAME,
+    REPORT_PATH,
+    REQUEST_BATCH_SIZE
+)
+from .classes import (
+    Course,
+    Block
+)
+
+FETCH_ALL_COURSES_URL = f"{LMS_URL}/api/courses/v1/courses/"
+
+def fetch_all_course_blocks(course: Course, cookie_jar):
+    '''
+    Fetch list of all course blocks in a course. Include the children blocks if present.
+    '''
+    fetch_blocks_url = f"{course.blocks_url}&all_blocks=true&depth=all&requested_fields=children"
+
+    reqs = [grequests.get(fetch_blocks_url, cookies=cookie_jar)]
+    r = grequests.map(reqs)[0]
+    
+    blocks = r.json().get('blocks')
+
+    for key in blocks:
+        block = Block(blocks.get(key))
+        course.add_block(block)
+
+def fetch_all_active_courses(cookie_jar):
+    '''
+    Fetch list of all courses.
+    Discards courses whose end date is defined and is in the past.
+    Returns the other courses as active courses.
+    '''
+    active_courses = []
+
+    print("Fetching list of all courses")
+    reqs = [grequests.get(FETCH_ALL_COURSES_URL, cookies=cookie_jar)]
+    r = grequests.map(reqs)[0]
+    
+    courses = r.json().get('results')
+    today = datetime.now()
+    for course in courses:
+        course_active = False
+        end_date_str = course.get('end', None)
+        if end_date_str:
+            end_date_str = end_date_str.split('T')[0]
+            end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
+            if end_date >= today:
+                course_active = True
+        else:
+            course_active = True
```

Comment:
Life's too short to parse dates manually. Just install [python-dateutil](https://pypi.org/project/python-dateutil/), import `from dateutil.parser import parse`, and do something like:
```suggestion
        try:
            course_active = parse(end_date, ignoretz=True) >= today:
        except TypeError:
            # End date is undefined.
            course_active = True
```